Handle multiple CVEs per issue in official CVE feed

robert-cronin opened the pull request:
Fix multiple CVE handling in official CVE feed

Fixes https://github.com/kubernetes/website/issues/47003

This change modifies the script used to generate the CVE feed to correctly handle issues containing multiple CVEs. Key updates:

- Create separate entries for each CVE ID when multiple are present in a single issue
- Maintain original structure and behavior for the first CVE entry
- Ensure compatibility with existing feed consumers

This approach aims to resolve the malformed GUID issue while preserving the feed's integrity. Feedback and suggestions for improvement are welcome.

Here is the output from before and after the change:

[before.txt](https://github.com/user-attachments/files/16029574/before.txt)
[after.txt](https://github.com/user-attachments/files/16029570/after.txt)

robert-cronin commented:
Here is the output from before and after the change:

[before.txt](https://github.com/user-attachments/files/16029574/before.txt)
[after.txt](https://github.com/user-attachments/files/16029570/after.txt)

PushkarJ commented:
/hold

(Discussing more on possible solutions in linked issue)

PushkarJ commented on sig-security-tooling/cve-feed/hack/fetch-official-cve-feed.py:
```diff
@@ -91,10 +91,20 @@ def getCVEStatus(state, state_reason):
+                additional_cve = cve.copy()
+                additional_cve['id'] = additional_cve_id
+                additional_cve['external_url'] = f'https://www.cve.org/cverecord?id={additional_cve_id}'
+                additional_cve['_kubernetes_io'] = cve['_kubernetes_io'].copy()
```
Is this needed, since we copy `cve` dict in its entirety already?

robert-cronin replied:
I believe this is needed, the copy command makes a shallow copy. Without shallow copying the nested dictionary, the command to update the url updates the keys in both dictionaries. I've replaced both copy commands with a single deepcopy for brevity :+1:

PushkarJ reviewed (commented):
Few minor comments, I think this is okay to merge once these are addressed. I am also convinced this is the right approach to resolve this, so no issues in terms of revisiting that. Shared more here: https://github.com/kubernetes/website/issues/47003#issuecomment-2197708530

robert-cronin commented on sig-security-tooling/cve-feed/hack/fetch-official-cve-feed.py:
```diff
@@ -91,10 +91,20 @@ def getCVEStatus(state, state_reason):
-            cve['id'] = cve_id
-            cve['external_url'] = f'https://www.cve.org/cverecord?id={cve_id}'
-            cve['_kubernetes_io']['google_group_url'] = f'https://groups.google.com/g/kubernetes-announce/search?q={cve_id}'
+            cve_ids = title[0].split(', ')
```
Great idea, this should handle the case when there is no space after the comma

robert-cronin commented:
@PushkarJ I've gone through and updated the script based on the feedback received. Here's the updated output:

[after2.txt](https://github.com/user-attachments/files/16038730/after2.txt)

Thank you for your suggestions!

PushkarJ commented:
@robert-cronin Thank you so much for your efforts in fixing this and addressing the comments and coming up with even better idea about dictionary copy

/lgtm
/approve

PushkarJ commented:
/hold cancel

robert-cronin commented:
> @robert-cronin Thank you so much for your efforts in fixing this and addressing the comments and coming up with even better idea about dictionary copy
> 
> /lgtm
> /approve
> 
> 
No problems, happy to help 🙂
